src/expr/src/scalar/func.rs

```diff
@@ -2066,6 +2066,30 @@ fn parse_ident<'a>(
     })?)
 }
 
+fn regexp_split_to_array<'a>(
+    text: Datum<'a>,
+    regexp: Datum<'a>,
+    flags: Datum<'a>,
+    temp_storage: &'a RowArena,
+) -> Result<Datum<'a>, EvalError> {
+    let text = text.unwrap_str();
+    let regexp = regexp.unwrap_str();
+    let flags = flags.unwrap_str();
+    let regexp = build_regex(regexp, flags)?;
```

Comment:
From the Regex crate's documentation:


> Note that regex compilation tends to be a somewhat expensive process, and unlike higher level environments, compilation is not automatically cached for you. One should endeavor to compile a regex once and then reuse it. For example, it’s a bad idea to compile the same regex repeatedly in a loop.

I imagine it will be very common that `regexp` is a literal and so we can optimize this by only constructing the regex object once. Can you file an issue to support this optimization?

Reply:
Btw. we have this optimization on RegexpMatch here: https://github.com/MaterializeInc/materialize/blob/main/src/expr/src/scalar/mod.rs#L1132

Reply:
Yup, this will be a followup PR.